blockindex/contractstaking/indexer.go

```diff
@@ -56,13 +56,16 @@ type (
 )
 
 // NewContractStakingIndexer creates a new contract staking indexer
-func NewContractStakingIndexer(kvStore db.KVStore, contractAddr string, contractDeployHeight uint64) *Indexer {
+func NewContractStakingIndexer(kvStore db.KVStore, contractAddr string, contractDeployHeight uint64) (*Indexer, error) {
+	if _, err := address.FromString(contractAddr); err != nil {
+		return nil, errors.Wrapf(err, "invalid contract address %s", contractAddr)
+	}
```

Comment:
can simply `panic("invalid system staking contract address")`, then no need to change other places

Reply:
also 
```
if kv == nil {
		panic("nil kvstore")
	}
```

Reply:
I think that it is generally recommended to return an error in most error handling scenarios instead of panic. Excessive use of panic can make the code less maintainable and harder to write unit tests.

Reply:
ok, then check the `kv == nil` error?

Reply:
ok